proparse/src/main/java/org/prorefactor/proparse/antlr4/PostLexer.java

```diff
@@ -239,38 +256,50 @@ private boolean preproIfCond(boolean evaluate) throws IOException, TokenStreamEx
     if (tokenVector.isEmpty() || !evaluate)
       return false;
     else {
-      DoParse evalDoParse = new DoParse(doParse.getRefactorSession(), null, doParse);
-      evalDoParse.preProcessCondition = true;
-      for (int i = 0; i < 4; i++) {
-        tokenVector.add(new ProToken(filenameList, ProParserTokenTypes.EOF, ""));
-      }
-      try {
-        evalDoParse.doParse(tokenVector);
-      } catch (ProEvalException e) {
-        String str = "Unable to evaluate &IF condition:";
-        for (ProToken tok : tokenVector) {
-          str += " " + tok.getText();
-        }
-        String fileName = null;
-        if (doParse.isValidIndex(currToken.getFileIndex()))
-          fileName = doParse.getFilename(currToken.getFileIndex());
-        throw new ProEvalException(str, e, fileName, currToken.getLine(), currToken.getColumn());
-      }
-      return evalDoParse.preProcessConditionResult;
+      CommonTokenStream cts = new CommonTokenStream(new ListTokenSource(tokenVector));
+      PreprocessorParser parser = new PreprocessorParser(cts);
+      return eval.visitPreproIfEval(parser.preproIfEval());
     }
   }
 
   private void throwMessage(String theMessage) {
-    int theIndex = currToken.getFileIndex();
-    if (doParse.isValidIndex(theIndex))
-      throw new IllegalArgumentException(doParse.getFilename(theIndex) + ":" + currToken.getLine() + " " + theMessage);
-    else
-      throw new IllegalArgumentException(theMessage);
+    // TODO Add file name in message
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1135)